evalml/tests/automl_tests/test_automl.py

```diff
@@ -2251,6 +2257,65 @@ def test_automl_pipeline_random_seed(mock_fit, mock_score, random_seed, X_y_mult
             assert automl.get_pipeline(row['id']).random_seed == random_seed
 
 
+@pytest.mark.parametrize("ensembling", [True, False])
+@patch('evalml.automl.automl_search.split_data')
+@patch('evalml.pipelines.BinaryClassificationPipeline.score', return_value={"Log Loss Binary": 0.3})
+@patch('evalml.pipelines.BinaryClassificationPipeline.fit')
+def test_automl_ensembling_training(mock_fit, mock_score, mock_split_data, ensembling, X_y_binary):
+    X, y = X_y_binary
+    X_train, X_ensemble, y_train, y_ensemble = split_data(X, y, problem_type='binary', test_size=0.2)
+    mock_split_data.return_value = (X_train, X_ensemble, y_train, y_ensemble)
+    # don't train the best pipeline since we check usage of the ensembling CV through the .fit mock
+    automl = AutoMLSearch(X_train=X, y_train=y, problem_type='binary', random_state=0, n_jobs=1, max_batches=19, ensembling=ensembling, train_best_pipeline=False, optimize_thresholds=False)
+    automl.search()
+    if ensembling:
+        assert automl.ensembling
+        # check that the X_train data is all used for the length
+        assert len(X_train) == (len(mock_fit.call_args_list[-2][0][0]) + len(mock_score.call_args_list[-2][0][0]))
+        # last call will be the stacking ensembler
```

Comment:
I don't think this will be true when we add another estimator to AutoMLSearch, e.g. KnnClassifier.

Maybe we should make max_batches depend on the number of allowed estimators for binary problems.